### PR TITLE
MEN-3478: update inventory test to verify software versions

### PR DESCRIPTION
**Pull request by tranchitella**

### CODE REVIEW

File: tests/tests/test_inventory.py [hunk 1]
```diff
@@ -13,39 +13,70 @@
...
     def test_inventory(self):
-        """Test that device reports inventory after having bootstrapped."""
-
-        attempts = 10
-
-        while True:
-            attempts = attempts - 1
+        """
+        Test that device reports inventory after having bootstrapped and performed
+        an application update using a dummy script artifact.
```
Comment (kacf) [inline tests/tests/test_inventory.py:42]: Aha! The [famous bootstrap issue](https://tracker.mender.io/browse/MEN-2582). It didn't occur to me that we need that here.

File: tests/tests/test_inventory.py [hunk 2]
```diff
@@ -13,39 +13,70 @@
...
+        """
+
+        # create a simple artifact (script) which doesn't do anything
+        artifact_name = "simple-artifact"
+        with tempfile.NamedTemporaryFile() as tf:
+            artifact = make_script_artifact(
+                artifact_name,
+                conftest.machine_name,
+                tf.name,
+                extra_args="--software-name swname --software-version v1",
```
Comment (kacf) [inline tests/tests/test_inventory.py:52]: I can't see that this is checked anywhere. Did you forget to actually modify the pass condition after you made the bootstrapping installation? :-)

File: tests/tests/test_inventory.py [hunk 3]
```diff
@@ -127,17 +158,13 @@ def test_inventory(self):
...
-                    time.sleep(5)
-                    continue
-                else:
-                    raise
+            except Exception as e:
+                latest_exception = e
+                time.sleep(5)
+            else:
+                return
+        raise last_exception
```
Comment (kacf) [inline tests/tests/test_inventory.py:170]: I think it should be `latest_exception`, no?

**Review by kacf — CHANGES_REQUESTED** [review]
Remember to target `feature-partial_updates` branch.

**Comment by tranchitellawfp** [comment]
@kacf sorry, this is a draft and not ready for review

**Comment by tranchitella** [comment]
@kacf sorry, this is a draft, not ready for review

File: tests/tests/test_inventory.py (hunk 3 above)
Comment (tranchitella) [reply tests/tests/test_inventory.py]: fixed

**Comment by tranchitella** [comment]
Pipeline launched manually:
https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/196175521

File: tests/tests/test_inventory.py (hunk 2 above)
Comment (tranchitella) [reply tests/tests/test_inventory.py:54]: I've added the check

**Comment by tranchitella** [comment]
@kacf ready for review

File: tests/tests/test_inventory.py [hunk 4]
```diff
@@ -86,6 +119,12 @@ def test_inventory(self):
                             )
                             in attrs
                         )
+                        assert (
+                            json.loads(
+                                '{"name": "rootfs-image_swname_version", "value": "v1"}'
```
Comment (kacf) [inline tests/tests/test_inventory.py:124]: We will need to fix this somehow (discussed in chat).

**Review by kacf — CHANGES_REQUESTED** [review]
Still wrong target branch.

**Comment by tranchitella** [comment]
Depends on https://github.com/mendersoftware/inventory/pull/213
Manually launched pipeline: https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/196342866

**Review by kacf — APPROVED** [review]

File: tests/tests/test_inventory.py (hunk 4 above)
Comment (tranchitella) [reply tests/tests/test_inventory.py]: Fixed

**Pull request merged**